adcc/LazyMp.py

```diff
@@ -122,9 +160,24 @@ def dipole_moment(self, level=2):
             raise NotImplementedError("Only dipole moments for level 1 and 2"
                                       " are implemented.")
 
+    @cached_member_function
     def energy_correction(self, level=2):
         """Obtain the MP energy correction at a particular level"""
-        return self.__cpp.energy_correction(level)
+        if level > 3:
+            raise NotImplementedError(f"MP({level}) energy correction "
+                                      "not implemented.")
+        if level < 2:
+            return 0.0
+        hf = self.reference_state
+        if level == 2 and not self.has_core_occupied_space:
+            return compute_energy_correction(hf.oovv, self.t2oo)
+        elif level == 2 and self.has_core_occupied_space:
+            pairs = [(self.t2oo, hf.oovv),
+                     (2.0 * self.t2oc, hf.ocvv),
+                     (self.t2cc, hf.ccvv)]
```

Comment:
This again feels like something which you should be able to unify if you loop over spaces (and a coefficient). In that way you can also scratch the far too simple `compute_energy_correction` function.

Reply:
Done.